custom/samveg/case_importer/operations.py

```diff
@@ -7,5 +12,6 @@ def run(cls, *args, **kwargs):
 class AddCustomCaseProperties(BaseOperation):
     @classmethod
     def run(cls, row_num, raw_row, fields_to_update):
-        # ToDo: Add new case property
+        today = datetime.utcnow().astimezone(pytz.timezone('Asia/Kolkata')).date()
```

Comment:
It seems confusing to store this in the project time zone when other system case properties will be UTC. Is this definitely going to show the expected time zone in UIs like the case list explorer and the case_data page? I thought we had reports code that detected datetime case properties and converted them to the project's time zone, which would risk double-converting this value.

Reply:
Hey @orangejenny 

> when other system case properties will be UTC

Are you referring to case meta properties?

> Is this definitely going to show the expected time zone in UIs like the case list explorer and the case_data page

The project is in India, so I don't think we need to worry about timezone here really. And I think the plan is to use this in the app somewhere. Even if its in reports, the user would be in IST timezone.
Plus, we can't have timezone aware date, so we would need to use datetime if we need timezone present, so we could store `str(datetime.utcnow())` instead and then the app would need to parse that into a datetime and extract the day. Is that possible in the app?

> I thought we had reports code that detected datetime case properties

I didn't know about this. Can you share one instance? But even if that is true, users for this project are in IST but if we can store UTC and then app can find the date, we can store UTC datetime instead.

Reply:
Yeah, I mean other meta properties. If a dev needs to troubleshoot something and is looking at the case in the db, they're likely going to assume that any value from HQ will be UTC. It's less of an issue for dates than datetimes, but we have had timezone-related date issues. Timezone bugs are so awful that I'm inclined to be very standard about using UTC for absolutely everything on the server. So I'd be in favor of making this property a datetime and storing in UTC, then letting the app builders convert to IST.

[Here's](https://github.com/dimagi/commcare-hq/blob/aaa9a859a793bd832ab2d524d03a0b3617a4740a/corehq/apps/hqwebapp/templatetags/proptable_tags.py#L189-L193) the kind of code I was thinking of (dealing with datetimes, not dates).

Reply:
offlined to keep date itself, considering the use case and project being in IST